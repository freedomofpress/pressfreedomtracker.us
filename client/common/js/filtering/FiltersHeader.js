import React, { PureComponent } from 'react'
import FilterSummary from '~/filtering/FilterSummary'
import { SettingsIcon } from '~/filtering/Icons'


class FiltersHeader extends PureComponent {
	render() {
		const {
			categories,
			categoriesEnabled,
			changeFiltersMessage,
			filtersExpanded,
			filterValues,
			handleToggle,
		} = this.props


		return (
			<div className="filters__header">
				<FilterSummary
					filtersExpanded={filtersExpanded}
					filterValues={filterValues}
					categories={categories}
					categoriesEnabled={categoriesEnabled}
					changeFiltersMessage={changeFiltersMessage}
				/>

				<button
					className="filters__button filters__button--summary-toggle"
					onClick={handleToggle}
					type="button"
				>
					<SettingsIcon />

					{filtersExpanded ? 'Collapse Filters' : 'Change Filters'}
				</button>
			</div>
		)
	}
}


export default FiltersHeader
