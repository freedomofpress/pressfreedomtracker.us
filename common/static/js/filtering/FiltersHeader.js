import React, { PureComponent } from 'react'
import FilterSummary from '~/filtering/FilterSummary'
import { SettingsIcon } from '~/filtering/Icons'


class FiltersHeader extends PureComponent {
	render() {
		const {
			filtersExpanded,
			filterValues,
			handleToggle,
			categoriesEnabled,
			changeFiltersMessage,
		} = this.props


		return (
			<div className="filters__header">
				<FilterSummary
					filtersExpanded={filtersExpanded}
					filterValues={filterValues}
					categoriesEnabled={categoriesEnabled}
					changeFiltersMessage={changeFiltersMessage}
				/>

				<button
					className="filters__button filters__button--summary-toggle"
					onClick={handleToggle}
				>
					<SettingsIcon />

					{filtersExpanded ? 'Collapse Filters' : 'Change Filters'}
				</button>
			</div>
		)
	}
}


export default FiltersHeader
