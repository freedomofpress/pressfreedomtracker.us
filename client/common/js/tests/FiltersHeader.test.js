import React from 'react';
import { shallow, mount } from 'enzyme';
import FiltersHeader from '~/filtering/FiltersHeader';
import {
	CATEGORIES_FILTER,
	CATEGORIES_ENABLED,
    FILTER_VALUES
} from '~/tests/factories/CategoriesFilterFactory';

describe('FiltersHeader', () => {
	let categories;
	let categoriesEnabled;
	let filterValues;

    beforeEach(() => {
        categories = CATEGORIES_FILTER;
        categoriesEnabled = CATEGORIES_ENABLED;
        filterValues = FILTER_VALUES;
    });

    afterEach(() => {
        categories = null;
        categoriesEnabled = null;
        filterValues = null;
    });

    it('should render correctly in "debug" mode', () => {
        const component = shallow(
        <FiltersHeader
            categories={categories}
            debug
        />
        );

        expect(component).toMatchSnapshot();
	});

	it('should render FiltersHeader properly with button toggle', () => {
		const component = mount(
        <FiltersHeader
            categories={categories}
            categoriesEnabled={categoriesEnabled}
            filterValues={filterValues}
        />
        );
        const filterButton = component.find('button.filters__button--summary-toggle');
        expect(filterButton.text()).toEqual('Change Filters');
        expect(filterButton.find('svg.filters__icon').length).toEqual(1);
	});

	it('should show FilterSummary properly according to filters', () => {
		const component = mount(
            <FiltersHeader
                categories={categories}
                categoriesEnabled={categoriesEnabled}
                filterValues={filterValues}
            />
		);
		const filterSummary = component.find('div.filters__summary');
		// 1 Category List, 1 Filter List = 2 List
		expect(filterSummary.find('ul.filters__summary-list').length).toEqual(2);
		expect(filterSummary.find('div.filters__text--dim').text()).toEqual(
			'Showing Leak Case Charged under espionage act?: Yes'
		);
	});
});
