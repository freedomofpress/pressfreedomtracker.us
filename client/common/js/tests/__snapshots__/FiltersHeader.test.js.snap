// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersHeader should render without error in "debug" mode 1`] = `ShallowWrapper {}`;
